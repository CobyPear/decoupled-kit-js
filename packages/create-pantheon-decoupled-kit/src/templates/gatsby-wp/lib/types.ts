import { type CreatePagesArgs } from 'gatsby';

interface QueryType<Type> {
	edges: {
		node: Type;
	}[];
}

type PostKeys = 'id' | 'title' | 'excerpt';

export type Post = {
	[key in PostKeys]: string;
};

export interface PostsQuery {
	posts: QueryType<Post>;
}

export type PrivatePosts = {
	id: string;
};

export interface PrivatePostsQuery {
	posts: QueryType<PrivatePosts>;
}

export interface GatsbyGraphQLHelper {
	graphql: CreatePagesArgs['graphql'];
	reporter: CreatePagesArgs['reporter'];
}
