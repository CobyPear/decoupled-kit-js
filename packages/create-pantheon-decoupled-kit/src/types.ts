import type { Answers, QuestionCollection } from 'inquirer';
import type { ParsedArgs } from 'minimist';
import type { SpyInstance } from 'vitest';

declare module 'vitest' {
	export interface TestContext {
		[key: string]: SpyInstance;
	}
}

/**
 * Generators need prompts to get user data not provided by CLI arguments
 */
export interface DecoupledKitGenerator<Prompts extends Answers> {
	/**
	 * Generator's name. This should be kebab case.
	 */
	name: string;
	/**
	 * Description of the generator
	 */
	description: string;
	/**
	 * An array of inquirer prompts
	 */
	prompts: QuestionCollection<Prompts>[];
	/**
	 * An array of paths to the generator's templates.
	 * This should be empty if the generator does not have templates.
	 */
	templates: string[];
	/**
	 * An array of actions to run with the prompts and templates
	 */
	actions: Action[];
	/**
	 * Any extra data that should be passed from the generator to the actions
	 */
	data?: DataRecord;
	/**
	 * Set to true if the generator is considered an addon.
	 * This will give priority to the templates when de-duping.
	 */
	addon?: boolean;
}

/**
 * An action that takes in the data, templates, and an instance of handlebars
 * and does an action, like installing dependencies or formatting generated code
 */
export type Action = (config: ActionConfig) => Promise<string> | string;

export type ActionRunner = (config: ActionRunnerConfig) => Promise<string>;

/**
 * Input from command line arguments and/or prompts
 */
export type Input = ParsedArgs & Answers;

export interface TemplateData {
	templateDirs: string[];
	addon: boolean;
}

export interface MergedPaths {
	[key: string]: { addon: boolean; base: string };
}
interface ActionConfig {
	data: Input;
	templateData: TemplateData[];
	handlebars: typeof Handlebars;
}

interface ActionRunnerConfig extends ActionConfig {
	actions: Action[];
}

type DataMember = string | number | boolean;
type DataRecord = {
	[key: string]: DataMember | Record<string, DataRecord>;
};

// TYPE PREDICATES

/**
 * @param arg a variable
 * @returns true if the variable is a string, false otherwise
 */
export const isString = (arg: unknown): arg is string => {
	if (typeof arg === 'string') {
		return true;
	}
	return false;
};
