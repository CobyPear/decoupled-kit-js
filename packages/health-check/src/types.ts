type CMSEndpointObjectBooleans<Key> = {
	[key in Key extends 'isValid' ? 'isValid' : 'hasUmami']: boolean;
};
type CMSEndpointObjectStrings = {
	[key in 'envVar' | 'endpoint']: string;
};

export type CMSEndpointObject<BooleanKey> = CMSEndpointObjectStrings &
	CMSEndpointObjectBooleans<BooleanKey>;
