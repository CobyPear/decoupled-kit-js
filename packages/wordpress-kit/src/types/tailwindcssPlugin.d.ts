import type { Config } from 'tailwindcss';

export type ColorConfig = {
  primary?: string;
  secondary?: string;
  tertiary?: string;
  foreground?: string;
  cyanBluishGray?: string;
  palePink?: string;
  vividRed?: string;
  luminousVividOrange?: string;
  luminousVividAmber?: string;
  lightGreenCyan?: string;
  vividGreenCyan?: string;
  paleCyanBlue?: string;
  vividCyanBlue?: string;
  vividPurple?: string;
  darkGray?: string;
  lightGray?: string;
  white?: string;
};

export type PaddingConfig = {
  backgroundX?: string;
  backgroundY?: string;
};

export type FontSizeConfig = {
  '7xl'?: string;
  '4xl'?: string;
  xl?: string;
  sm?: string;
};

export type WordpressMapConfig = {
  theme?: {
    extend?: {
      /**
       * The colors mapped from the code editor in the WordPress admin.
       * @example
       * ```
       *primary: '#0070f3',
       *secondary: '#ff4081',
       *darkGray: '#333',
       *lightGray: '#fafafa',
       *white: '#fff',
       * ```
       *
       *@default
       * ```
       *primary: value of "blue-500" or '#0073a8',
       *secondary: value of "gray-500" or '#005075',
       *darkGray: value of "neutral-800" or '#333333',
       *lightGray: value of "neutral-500" or '#666666',
       *white: value of "white" or '#ffffff',
       * ```
       */
      colors?: ColorConfig;
      /**
       * Padding applied when the block has a background color.
       * @example
       * ```
       *backgroundX: '2rem',
       *backgroundY: '1rem',
       * ```
       *
       * @default
       * ```
       * backgroundX: value of p-5 or '1.25rem',
       * backgroundY: value of p-9 or '2.35rem',
       * ```
       */
      padding?: PaddingConfig;
      /**
       * The font size map for the code editor in the WordPress admin.
       *
       * ```
       * huge: text-7xl
       * large: text-4xl
       * normal: text-xl
       * small: text-sm
       *```
       *
       * @example
       * ```
       *"7xl": 5rem;
       *"4xl": 4rem;
       *xl: 2rem;
       *sm: 1rem;
       * ```
       *
       * @default
       * ```
       * "7xl": '5rem',
       * "4xl": '4rem',
       * xl: '1.25rem',
       * sm: '1rem',
       */
      fontSize?: FontSizeConfig;
    };
    /**
     * The colors mapped from the code editor in the WordPress admin.
     * @example
     * ```
     *primary: '#0070f3',
     *secondary: '#ff4081',
     *darkGray: '#333',
     *lightGray: '#fafafa',
     *white: '#fff',
     * ```
     *
     *@default
     * ```
     *primary: value of "blue-500" or '#0073a8',
     *secondary: value of "gray-500" or '#005075',
     *darkGray: value of "neutral-800" or '#333333',
     *lightGray: value of "neutral-500" or '#666666',
     *white: value of "white" or '#ffffff',
     * ```
     */
    colors?: ColorConfig;
    /**
     * Padding applied when the block has a background color.
     * @example
     * ```
     *backgroundX: '2rem',
     *backgroundY: '1rem',
     * ```
     *
     * @default
     * ```
     * backgroundX: value of p-5 or '1.25rem',
     * backgroundY: value of p-9 or '2.35rem',
     * ```
     */
    padding?: PaddingConfig;
    /**
     * The font size map for the code editor in the WordPress admin.
     *
     * ```
     * huge: text-7xl
     * large: text-4xl
     * normal: text-xl
     * small: text-sm
     *```
     *
     * @example
     * ```
     *"7xl": 5rem;
     *"4xl": 4rem;
     *xl: 2rem;
     *sm: 1rem;
     * ```
     *
     * @default
     * ```
     * "7xl": '5rem',
     * "4xl": '4rem',
     * xl: '1.25rem',
     * sm: '1rem',
     */
    fontSize?: FontSizeConfig;
  };
};

export type Color = {
  name: string;
  themeName: string;
  tailwindDefault: string;
  hexDefault: string;
};

export type FontSize = {
  name: string;
  tailwind: string;
  default: string;
};

export type TailwindcssConfig = Config & WordpressMapConfig;
