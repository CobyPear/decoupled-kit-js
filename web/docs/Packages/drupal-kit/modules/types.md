---
id: "types"
title: "Module: types"
sidebar_label: "types"
sidebar_position: 0
custom_edit_url: null
---

## Interfaces

- [authInit](../interfaces/types.authInit.md)
