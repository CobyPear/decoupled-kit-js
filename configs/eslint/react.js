/** @type {import('eslint').ESLint.ConfigData} */
module.exports = {
	plugins: ['react'],
	extends: [
		'./index.js',
		'plugin:react/recommended',
		'plugin:react-hooks/recommended',
		'plugin:react/jsx-runtime',
	],
	settings: {
		react: {
			version: 'detect',
		},
	},
};
