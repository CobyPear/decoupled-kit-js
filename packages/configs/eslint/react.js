module.exports = {
	plugins: ['react'],
	extends: [
		'./index.js',
		'plugin:react/recommended',
		'plugin:react-hooks/recommended',
	],
	settings: {
		react: {
			version: 'detect',
		},
	},
};
