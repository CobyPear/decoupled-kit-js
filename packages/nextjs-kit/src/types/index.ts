export interface DataToSort {
  data: Record<string, string | number>[];
  key: string;
  direction: 'asc' | 'ASC' | 'desc' | 'DESC';
}
export interface LinkProps {
  linkText: string;
  href: string;
}

type Parent = 'parent' | 'parentId';

export type FooterMenuItem = LinkProps & {
  [key in Parent]?: string | null;
};

/**
 * Type predicate to determine if a FooterMenuItem has a parent or parentId
 * @param {FooterMenuItem} item a `FooterMenuItem`
 * @returns true if `parentId` or `parent` properties are found on the `FooterMenuItem`
 */
export const hasParent = (item: FooterMenuItem): item is FooterMenuItem =>
  item.parentId ? true : item.parent ? true : false;

export const isNumber = (item: number | null | undefined): item is number =>
  typeof item === 'number';
export interface PaginatorObj {
  id: string;
  title: string;
  body: {
    value: string;
  };
}

export interface Pagination {
  data: PaginatorObj[];
  itemsPerPage: number;
  breakpoints: {
    start: number;
    end: number;
    add: number;
  };
  routing: boolean;
  Component: React.ElementType;
}
