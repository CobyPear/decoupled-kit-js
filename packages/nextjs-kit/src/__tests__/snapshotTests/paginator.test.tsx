import { render } from '@testing-library/react';
import Paginator from '../../components/paginator';
import examplePaginationData from '../data/examplePaginationData.json';
import { PaginatorObj } from '../../types/index';
import React from 'react';
import { vi } from 'vitest';

vi.mock('next/router', () => ({
  useRouter: () => ({
    locale: 'en',
    pathname: 'test/path',
    push: vi.fn(),
    query: {
      page: '/examples/pagination/[[...page]]',
    },
  }),
}));

/**
 * @vitest-environment jsdom
 */

interface PaginationItemProps {
  currentItems: PaginatorObj[];
}
const RenderCurrentItems: React.FC<PaginationItemProps> = ({
  currentItems,
}: PaginationItemProps): JSX.Element => {
  return (
    <React.Fragment>
      {currentItems.map(item => {
        return (
          <article
            key={item.id}
            className="flex flex-col p-3 w-fit mx-auto mb-10"
          >
            <h2 className="justify-start my-auto text-2xl mb-2">
              {item.title}
            </h2>
            <p className="max-w-prose my-2">
              {item?.body.value.substr(0, 150)}...
            </p>
          </article>
        );
      })}
    </React.Fragment>
  );
};
describe(`<Paginator />`, () => {
  it('should render the paginated data', () => {
    const { asFragment } = render(
      <div className="prose container min-w-full min-h-screen max-w-screen mx-auto">
        <main className="flex mx-auto flex-col">
          <section className="mx-auto">
            <h1 className="my-10">Pagination example</h1>
            <Paginator
              data={examplePaginationData}
              itemsPerPage={10}
              breakpoints={{ start: 6, end: 12, add: 6 }}
              routing={true}
              Component={RenderCurrentItems}
            />
          </section>
        </main>
      </div>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
