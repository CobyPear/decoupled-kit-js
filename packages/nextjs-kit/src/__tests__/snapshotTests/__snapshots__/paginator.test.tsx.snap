// Vitest Snapshot v1

exports[`<Paginator /> > should render the paginated data 1`] = `
<DocumentFragment>
  <div
    class="prose container min-w-full min-h-screen max-w-screen mx-auto"
  >
    <main
      class="flex mx-auto flex-col"
    >
      <section
        class="mx-auto"
      >
        <h1
          class="my-10"
        >
          Pagination example
        </h1>
        <div
          class="max-w-screen-md"
        >
          <h3
            class="mb-8 prose-sm"
          >
            Page 1/18
          </h3>
          <section>
            <article
              class="flex flex-col p-3 w-fit mx-auto mb-10"
            >
              <h2
                class="justify-start my-auto text-2xl mb-2"
              >
                Usitas
              </h2>
              <p
                class="max-w-prose my-2"
              >
                Comis defui ex ille lobortis molior premo ulciscor. Diam huic iustum mauris quia voco. Commodo ibidem iriure nibh nostrud olim scisco vereor. Conventi...
              </p>
            </article>
            <article
              class="flex flex-col p-3 w-fit mx-auto mb-10"
            >
              <h2
                class="justify-start my-auto text-2xl mb-2"
              >
                Comis Ex
              </h2>
              <p
                class="max-w-prose my-2"
              >
                Brevitas conventio cui dolor minim os. Amet lenis quibus. Adipiscing exerci ille immitto lucidus roto valetudo volutpat. Caecus causa dolore eros eum ...
              </p>
            </article>
            <article
              class="flex flex-col p-3 w-fit mx-auto mb-10"
            >
              <h2
                class="justify-start my-auto text-2xl mb-2"
              >
                Caecus Enim Macto Persto
              </h2>
              <p
                class="max-w-prose my-2"
              >
                Blandit erat luptatum praesent refoveo veniam. Bene distineo veniam. Eu facilisi feugiat ille. Molior nunc proprius turpis. Abdo eu similis. Accumsan ...
              </p>
            </article>
            <article
              class="flex flex-col p-3 w-fit mx-auto mb-10"
            >
              <h2
                class="justify-start my-auto text-2xl mb-2"
              >
                In
              </h2>
              <p
                class="max-w-prose my-2"
              >
                Adipiscing at ex hos. Ratis refero tamen. Amet interdico jus pala rusticus typicus. Gravis importunus interdico iriure pala paratus paulatim populus r...
              </p>
            </article>
            <article
              class="flex flex-col p-3 w-fit mx-auto mb-10"
            >
              <h2
                class="justify-start my-auto text-2xl mb-2"
              >
                Appellatio Eros Letalis
              </h2>
              <p
                class="max-w-prose my-2"
              >
                Conventio in interdico saluto vulputate. Camur commoveo elit feugiat macto meus nutus similis utinam vulputate. Paulatim premo usitas. Consectetuer er...
              </p>
            </article>
            <article
              class="flex flex-col p-3 w-fit mx-auto mb-10"
            >
              <h2
                class="justify-start my-auto text-2xl mb-2"
              >
                Dolus Iaceo Tum Verto
              </h2>
              <p
                class="max-w-prose my-2"
              >
                Jumentum neo quidne tincidunt ut. Iaceo ideo jugis velit. Adipiscing damnum euismod neque pertineo praemitto tation te valde valetudo. Capto enim leni...
              </p>
            </article>
            <article
              class="flex flex-col p-3 w-fit mx-auto mb-10"
            >
              <h2
                class="justify-start my-auto text-2xl mb-2"
              >
                Defui Exputo Gilvus Suscipit
              </h2>
              <p
                class="max-w-prose my-2"
              >
                Capto diam iaceo pertineo plaga quidem quidne suscipere ullamcorper virtus. Antehabeo distineo eu euismod iusto letalis natu sino suscipit. Appellatio...
              </p>
            </article>
            <article
              class="flex flex-col p-3 w-fit mx-auto mb-10"
            >
              <h2
                class="justify-start my-auto text-2xl mb-2"
              >
                Eros Iriure Venio Vulpes
              </h2>
              <p
                class="max-w-prose my-2"
              >
                Adipiscing conventio fere iaceo magna quidne sed. Facilisis ratis scisco uxor. Elit exerci ymo zelus. Comis ea nutus pala quae refero sudo te ut. Aute...
              </p>
            </article>
            <article
              class="flex flex-col p-3 w-fit mx-auto mb-10"
            >
              <h2
                class="justify-start my-auto text-2xl mb-2"
              >
                Eligo Praemitto
              </h2>
              <p
                class="max-w-prose my-2"
              >
                Erat haero iaceo nibh odio ratis roto. Aptent typicus valde. Aliquam macto melior quadrum suscipere. Facilisis fere nibh paulatim tego vicis.

Erat et...
              </p>
            </article>
            <article
              class="flex flex-col p-3 w-fit mx-auto mb-10"
            >
              <h2
                class="justify-start my-auto text-2xl mb-2"
              >
                Abluo Aliquam Inhibeo Luptatum
              </h2>
              <p
                class="max-w-prose my-2"
              >
                Abluo defui odio vel. Dolus ludus molior ullamcorper. Et letalis meus utrum uxor vel. Ad amet defui hos imputo praesent roto ut uxor. Augue erat eum l...
              </p>
            </article>
          </section>
          <div
            class="sticky lg:bottom-12 bottom-4"
          >
            <div
              class="flex flex-row justify-center mx-auto mt-auto mb-4"
            >
              <button
                class="h-16 w-12 disabled:bg-gray-500 hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 border-l-2 border-t-2 border-b-2 border-black bg-white"
                disabled=""
                id="back-btn"
              >
                &lt;
              </button>
              <button
                class="
          block
          h-16 w-12 border-t-2 border-b-2 border-black bg-white hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 border-blue-700 border-2
          "
              >
                1
              </button>
              <button
                class="
          hidden md:block
          h-16 w-12 border-t-2 border-b-2 border-black bg-white hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 
          "
              >
                2
              </button>
              <button
                class="
          hidden md:block
          h-16 w-12 border-t-2 border-b-2 border-black bg-white hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 
          "
              >
                3
              </button>
              <button
                class="
          hidden md:block
          h-16 w-12 border-t-2 border-b-2 border-black bg-white hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 
          "
              >
                4
              </button>
              <button
                class="
          hidden md:block
          h-16 w-12 border-t-2 border-b-2 border-black bg-white hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 
          "
              >
                5
              </button>
              <button
                class="hidden md:block h-16 w-12 border-2 border-black bg-slate-200 hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300\\"
            }"
              >
                ...
              </button>
              <button
                class="
          hidden md:block
          h-16 w-12 border-t-2 border-b-2 border-black bg-white hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 
          "
              >
                12
              </button>
              <button
                class="
          hidden md:block
          h-16 w-12 border-t-2 border-b-2 border-black bg-white hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 
          "
              >
                13
              </button>
              <button
                class="
          hidden md:block
          h-16 w-12 border-t-2 border-b-2 border-black bg-white hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 
          "
              >
                14
              </button>
              <button
                class="
          hidden md:block
          h-16 w-12 border-t-2 border-b-2 border-black bg-white hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 
          "
              >
                15
              </button>
              <button
                class="
          hidden md:block
          h-16 w-12 border-t-2 border-b-2 border-black bg-white hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 
          "
              >
                16
              </button>
              <button
                class="
          hidden md:block
          h-16 w-12 border-t-2 border-b-2 border-black bg-white hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 
          "
              >
                17
              </button>
              <button
                class="
          hidden md:block
          h-16 w-12 border-t-2 border-b-2 border-black bg-white hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300 
          "
              >
                18
              </button>
              <button
                class="h-16 w-12 disabled:bg-gray-500 hover:bg-blue-300 focus:bg-blue-200 focus:border-blue-300  border-r-2 border-t-2 border-b-2 border-black bg-white"
                id="next-btn"
              >
                &gt;
              </button>
            </div>
          </div>
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;
